sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -608,6 +673,39 @@ def _from_list(self, messages, parent_span=None):
         for message in messages:
             self._add(message, parent_span)
 
+    def _add(self, add_message, parent_span=None):
+        # type: (Union[ServiceBusMessage, Mapping[str, Any], AMQPAnnotatedMessage], AbstractSpan) -> None
+        """Actual add implementation.  The shim exists to hide the internal parameters such as parent_span."""
+        message = create_messages_from_dicts_if_needed(add_message, ServiceBusMessage)
+        message = transform_messages_to_sendable_if_needed(message)
+        message = cast(ServiceBusMessage, message)
+        trace_message(
+            message, parent_span
+        )  # parent_span is e.g. if built as part of a send operation.
+        message_size = (
+            message.message.get_message_encoded_size()
+        )
+
+        # For a ServiceBusMessageBatch, if the encoded_message_size of event_data is < 256, then the overhead cost to
+        # encode that message into the ServiceBusMessageBatch would be 5 bytes, if >= 256, it would be 8 bytes.
+        size_after_add = (
+            self._size
+            + message_size
+            + _BATCH_MESSAGE_OVERHEAD_COST[0 if (message_size < 256) else 1]
+        )
+
+        if size_after_add > self.max_size_in_bytes:
+            raise MessageSizeExceededError(
+                message="ServiceBusMessageBatch has reached its size limit: {}".format(
+                    self.max_size_in_bytes
+                )
+            )
+
+        self.message._body_gen.append(message)  # pylint: disable=protected-access
+        self._size = size_after_add
+        self._count += 1
+        self._messages.append(message)
```

Comment:
note: this part is just moving the private `_add` method in front of public methods, nothing changed.